### PR TITLE
Process data models

### PR DESCRIPTION
**Pull request by BonkeGcobo**
## Description
- [x] Populate the table of the blog database
- [x] Created the current user method in the application controller to return the first data from the user database

**Note**
I am not really sure if that's how  am I going to return data from the database in my Application controller,  also struggled to test if it works, I would really appreciate some insight

Happy Reviewing 😃

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts # user can have many posts
+  has_many :comments, through: :post # User can have 0 or more comments in one
+
```
Comment (mikethreels) [inline app/models/user.rb]: - [x] a User should have many likes as well.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts # user can have many posts
+  has_many :comments, through: :post # User can have 0 or more comments in one
```
Comment (mikethreels) [inline app/models/user.rb:3]: - [ ] you should set the foreign_key here as well

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments # Post can have 0 or more comments
+  has_many :likes # Posts can have 0 or more likes
+  has_many :users, through: :likes # Posts can have 0 or more likes from users
+  has_many :users, through: :comments # A posts has 0 or more users commenting on it
```
Comment (mikethreels) [inline app/models/post.rb]: - [x] a User doesn't belong to posts so Post doesn't have any Users.

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  def update_comments_counter
+    post.update(CommentsCounter: post.comments.count)
```
Comment (mikethreels) [inline app/models/comment.rb:6]: - [x] [OPTIONAL] you could use `increment!` here instead of update.

File: app/models/like.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  def update_likes
+    post.update(LikesCounter: post.likes.count)
```
Comment (mikethreels) [inline app/models/like.rb:6]: - [x] [OPTIONAL] you could use `increment!` here instead of update.

File: app/models/post.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+  def posts_counter
+    user.update(PostsCounter: user.posts.count)
```
Comment (mikethreels) [inline app/models/post.rb:7]: - [x] [OPTIONAL] you could use `increment!` here instead of update.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Bonke, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (BonkeGcobo) [reply app/models/user.rb:3]: Shouldn't the foreign key be found only in the corresponding migration file

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢🟢🟢 

Hi @BonkeGcobo, 

Good job working on the changes asked by the previous reviewer. 👍 
Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights
✅ Correct use of Github workflow
✅ No linter error
✅ README is fine

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**